apps/mull-api/src/app/event/event.service.ts

```diff
@@ -27,6 +27,13 @@ export class EventService {
     return this.findOne(eventInput.id);
   }
 
+  async addParticipant(eventId: number, userId: number) {
+    const user = new User(userId);
```

Comment:
Just a question, does `new User(userId)` create a new user in the DB? or is it like a `find_or_create`?

Reply:
Thank you! I have another question, what happens if the user doesn't exist in the database? Will that still get saved into the array?

Reply:
It would throw an error which would have to be handled by the caller of the mutation - on the front-end.

Reply:
By the time this function gets called, the user should already have registered, logged in and be authenticated, so they should already exist in the db. 

If somehow they manage to get past all that and provide an invalid userId, this function would throw an error which would have to be handled by the caller - the front-end.

Reply:
I'm not very familiar with graphQL so I might be totally wrong, but it's possible to query graphQL without having to go through the front-end right? I think for sanity check, it could be better if we find the user from the DB first( again I'm not very familiar with type ORM, maybe it's not possible to do that) the pseudo code would look like: 
```
user = User.find(id : id) 
if user:
  const event = await this.eventRepository.findOne(eventId, { relations: ['participants'] });
  event.participants.push(user);
 ```
What do you think?

Reply:
Yeah I know what you mean, I thought about that too. 
My concern was 3 things

1. That will make an additional db call, decreasing performance.
1. Now this function is doing multiple things (finding a user and then adding them to the event).
1. What should we do if a user is not found? Throw an error anyway right, so I don't see the point in doing an extra call to throw an error in the end when it will throw an error by default

Reply:
Ah, so type ORM actually checks if the user exists, thank you. I might be wrong again, and it really depends on if security is a concern for us, but if anyone can query the graphql api, then anyone can just post an event id and adding whatever user to the participant list...

Reply:
It won't let you add a user that does not exist since it is a foreign key and the user id needs to exist before adding them to an event.

Reply:
@ys-lin Definitely something to discuss in the future.
Maybe we can authenticate the user before running any mutations. However we would need the implementation of JTW in #99 for this. 
Definitely need more discussion about the security if our endpoint. That's a bigger issue than this ticket though. We can discuss how to handle that with our team.